fv/bpf_test.go

```diff
@@ -617,6 +617,160 @@ func describeBPFTests(opts ...bpfTestOpt) bool {
 					cc.CheckConnectivity()
 				})
 
+				It("should not be able to spoof workloads IP by another workload", func() {
+					if testOpts.protocol != "udp" {
+						return
+					}
+
+					felixes[0].EnsureBinary("pktgen")
+					felixes[1].EnsureBinary("pktgen")
+
+					By("allowing any traffic", func() {
+						pol.Spec.Ingress = []api.Rule{
+							{
+								Action: "Allow",
+								Source: api.EntityRule{
+									Nets: []string{
+										"0.0.0.0/0",
+									},
+								},
+							},
+						}
+						pol = updatePolicy(pol)
+
+						cc.ExpectSome(w[1][0], w[0][0])
+						cc.ExpectSome(w[1][1], w[0][0])
+						cc.CheckConnectivity()
+					})
+
+					/* XXX temporary will be removed and replaced by enfocing it
+					 * XXX from felix in BPF
+					 */
+					By("setting strict RPF check", func() {
+						felixes[1].Exec("sysctl", "-w", "net.ipv4.conf.all.rp_filter=1")
+						felixes[1].Exec("sysctl", "-w", "net.ipv4.conf.default.rp_filter=1")
+					})
+
+					By("testing that packet sent by another workload is dropped", func() {
+						tcpdump := w[0][0].AttachTCPDump()
+						tcpdump.SetLogEnabled(true)
+						matcher := fmt.Sprintf("IP %s.30444 > %s.30444: UDP", w[1][0].IP, w[0][0].IP)
```

Comment:
This is test code, so not super important but the IP and the format string both have `.` in them, which is a wildcard.  I think there's `regexp.QuoteMeta()` to do escaping where you want a literal match